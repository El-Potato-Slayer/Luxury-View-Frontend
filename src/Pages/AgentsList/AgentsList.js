import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

function AgentsList() {
  const { agents } = useSelector((state) => state.agentsReducer);

  function displayAgent(agent) {
    return (
      <div>
        <div
          className="listing-image"
          style={{ background: `url(${agent.picture}) center / cover` }}
        />
        <div className="info">
          <p className="listing-important">
            {agent.first_name}
            {' '}
            {agent.last_name}
          </p>
          <p>
            Number:
            {' '}
            {agent.number}
          </p>
          <p>
            Email:
            {' '}
            {agent.email}
          </p>
        </div>
      </div>
    );
  }

  return (
    <div data-testid="agents" className="page">
      <h2 className="page-title">Agents</h2>
      <div data-testid="agents-listing" className="listings">
        {agents.map((agent) => (
          <Link key={agent.id} to={`agents/${agent.id}`}>
            {displayAgent(agent)}
          </Link>
        ))}
      </div>
    </div>
  );
}

export default AgentsList;
