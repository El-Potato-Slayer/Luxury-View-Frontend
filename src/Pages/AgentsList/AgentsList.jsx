import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import SkeletonListAgent from '../../Skeletons/SkeletonListAgent/SkeletonListAgent';
import displayAgent from './helper';

function AgentsList() {
  const { agents, loading } = useSelector((state) => state.agentsReducer);

  return (
    <div data-testid="agents" className="page">
      <h2 className="page-title">Agents</h2>
      <div data-testid="agents-listing" className="listings">
        {loading && (
          <>
            <SkeletonListAgent />
            <SkeletonListAgent />
            <SkeletonListAgent />
          </>
        )}
        {agents.map((agent) => (
          <Link key={agent.id} to={`agents/${agent.id}`}>
            {displayAgent(agent)}
          </Link>
        ))}
      </div>
    </div>
  );
}

export default AgentsList;
