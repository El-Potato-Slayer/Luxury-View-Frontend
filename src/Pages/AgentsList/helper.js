export default function displayAgent(agent) {
  return (
    <div>
      <div
        className="listing-image"
        style={{ background: `url(${agent.picture}) center / cover` }}
      />
      <div className="info">
        <p className="listing-important">
          {agent.first_name}
          {' '}
          {agent.last_name}
        </p>
        <p>
          Number:
          {' '}
          {agent.number}
        </p>
        <p>
          Email:
          {' '}
          {agent.email}
        </p>
      </div>
    </div>
  );
}
