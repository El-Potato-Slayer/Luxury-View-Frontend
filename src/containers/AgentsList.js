import PropTypes from 'prop-types';
import { connect } from 'react-redux';

function AgentsList({ agents }) {
  return (
    <>
      <h2 data-testid="agents">Agents</h2>
      <div data-testid="agents-listing">
        {agents.map((agent) => (
          <p key={agent.id}>{agent.first_name}</p>
        ))}
      </div>
    </>
  );
}

const mapStateToProps = (state) => ({
  agents: state.agentsReducer.agents,
});

AgentsList.propTypes = {
  agents: PropTypes.instanceOf(Array).isRequired,
};

export default connect(mapStateToProps)(AgentsList);
