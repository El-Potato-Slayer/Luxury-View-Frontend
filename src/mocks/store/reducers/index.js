import { combineReducers } from 'redux';
import agentsReducer from './agentsReducer';
import appointmentsReducer from './appointmentsReducer';
import authReducer from './authReducer';
import propertiesReducer from './propertiesReducer';
import userReducer from './userReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  agentsReducer, appointmentsReducer, authReducer, propertiesReducer, userReducer, filterReducer,
});
