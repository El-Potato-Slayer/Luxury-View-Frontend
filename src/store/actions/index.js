export const setAgents = (payload) => ({
  type: 'SET_AGENTS',
  payload,
});

export const setAppointments = (payload) => ({
  type: 'SET_APPOINTMENTS',
  payload,
});

export const setProperties = (payload) => ({
  type: 'SET_PROPERTIES',
  payload,
});

export const changeFilter = (name) => ({
  type: 'CHANGE_FILTER',
  name,
});
