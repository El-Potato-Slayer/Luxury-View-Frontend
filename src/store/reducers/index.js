import { combineReducers } from 'redux';
import agentsReducer from './agentsReducer';
import appointmentsReducer from './appointmentsReducer';
import propertiesReducer from './propertiesReducer';
import userReducer from './userReducer';
import filterReducer from './filterReducer';

export default combineReducers({
  agentsReducer, appointmentsReducer, propertiesReducer, userReducer, filterReducer,
});
