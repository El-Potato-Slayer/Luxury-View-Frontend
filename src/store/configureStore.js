import { applyMiddleware, createStore } from 'redux';
import ReduxThunk from 'redux-thunk';
import { fetchData } from './actions';
import { fetchAgentsFailure, fetchAgentsRequest, fetchAgentsSuccess } from './actions/agentActions';
import { fetchAppointmentsFailure, fetchAppointmentsRequest, fetchAppointmentsSuccess } from './actions/appointmentActions';
import { fetchPropertiesFailure, fetchPropertiesRequest, fetchPropertiesSuccess } from './actions/propertyActions';
import rootReducer from './reducers';

const store = createStore(rootReducer, applyMiddleware(ReduxThunk));

// store.subscribe(() => { console.log(store.getState()); });
store.dispatch(fetchData('properties', fetchPropertiesRequest, fetchPropertiesSuccess, fetchPropertiesFailure));
store.dispatch(fetchData('agents', fetchAgentsRequest, fetchAgentsSuccess, fetchAgentsFailure));
store.dispatch(fetchData('appointments', fetchAppointmentsRequest, fetchAppointmentsSuccess, fetchAppointmentsFailure, true));

export default store;
