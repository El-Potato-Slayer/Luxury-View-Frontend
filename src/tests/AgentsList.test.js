import {
  cleanup, render, screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import AgentsList from '../containers/AgentsList';
import store from '../mocks/store/configureStore';

afterEach(cleanup);

test('Renders the AgentsList component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
    div,
  );
});

test('Checks that title is rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).toHaveTextContent('Agents');
});

test('Checks that agent names are rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).toHaveTextContent('Joshua Door');
  expect(screen.getByTestId('agents')).toHaveTextContent('Darlene');
  expect(screen.getByTestId('agents')).toHaveTextContent('Yapi Kobustus');
});

test('Checks that unknown agent names are not rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Ooga');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Booga');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Ooga Wooga Booga');
});

test('Checks that agent numbers are rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).toHaveTextContent('Number: 0987634391');
  expect(screen.getByTestId('agents')).toHaveTextContent('Number: 0987431489');
  expect(screen.getByTestId('agents')).toHaveTextContent('Number: 0874521321');
});

test('Checks that unknown agent numbers are not rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Number: 065748');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Number: 6584035224');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Number: 45');
});

test('Checks that agent emails are rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).toHaveTextContent('Email: joshuadoor@gmail.com');
  expect(screen.getByTestId('agents')).toHaveTextContent('Email: darthememphis@gmail.com');
  expect(screen.getByTestId('agents')).toHaveTextContent('Email: yapikobus@gmail.com');
});

test('Checks that unknown agent emails are not rendered', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  );
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Email: jok@gmail.com');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Email: yani@gmail.com');
  expect(screen.getByTestId('agents')).not.toHaveTextContent('Email: tollo@gmail.com');
});

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <AgentsList />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
