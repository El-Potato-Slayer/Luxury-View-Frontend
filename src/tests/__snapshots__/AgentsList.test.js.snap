// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="page"
  data-testid="agents"
>
  <h2
    className="page-title"
  >
    Agents
  </h2>
  <div
    className="listings"
    data-testid="agents-listing"
  >
    <a
      href="/agents/1"
      onClick={[Function]}
    >
      <div>
        <div
          className="listing-image"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/agents%2FP7JIqIE.png?alt=media&token=bfa90cd0-07e5-4b99-b15e-f678570049bd) center / cover",
            }
          }
        />
        <div
          className="info"
        >
          <p
            className="listing-important"
          >
            Joshua
             
            Door
          </p>
          <p>
            Number:
             
            0987634391
          </p>
          <p>
            Email:
             
            joshuadoor@gmail.com
          </p>
        </div>
      </div>
    </a>
    <a
      href="/agents/2"
      onClick={[Function]}
    >
      <div>
        <div
          className="listing-image"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/agents%2FEmbarrased_Man.jpg?alt=media&token=7168439b-a738-4122-96af-fba1dc158a9a) center / cover",
            }
          }
        />
        <div
          className="info"
        >
          <p
            className="listing-important"
          >
            Darlene (unisex name)
             
            Memphis
          </p>
          <p>
            Number:
             
            0987431489
          </p>
          <p>
            Email:
             
            darthememphis@gmail.com
          </p>
        </div>
      </div>
    </a>
    <a
      href="/agents/3"
      onClick={[Function]}
    >
      <div>
        <div
          className="listing-image"
          style={
            Object {
              "background": "url(https://firebasestorage.googleapis.com/v0/b/tactus-caelesti.appspot.com/o/agents%2Fsalesman.jpg?alt=media&token=0ec04d27-2070-4673-a6d0-bc0eb34cb10b) center / cover",
            }
          }
        />
        <div
          className="info"
        >
          <p
            className="listing-important"
          >
            Yapi
             
            Kobustus
          </p>
          <p>
            Number:
             
            0874521321
          </p>
          <p>
            Email:
             
            yapikobus@gmail.com
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
